calendar.rb

```diff
@@ -1,127 +1,125 @@
-        newevent=Event.new(title,desc)
-        event[date][index+1]=newevent
-        flag=true
-      end
+  def initialize(total_events)
+    @total_events = total_events
+  end
+  # add event
+  def add
+
+    user_input = input_for_add
+    return false unless user_input
+    
+    date, title, desc = user_input
+    flag = false
+    @total_events.each do |key, value|
+      next unless key == (date)
+
+      new_event = Event.new(title, desc)
+      value.push new_event
+      flag = true
     end
-    if(!flag)
-      event=Event.new(title,desc)
-      arr_of_events=[]
-      arr_of_events.push event
-      hash={date=>arr_of_events}
-      total_events<<hash
+    unless flag
+      new_event = Event.new(title, desc)
+      @total_events[date] = [new_event]
     end
+    p 'Event add successfully'
+    p @total_events
     true
   end
-  #Modify Event
-  def self.modify(total_events)
-    p "Enter date on which you modify event:"
-    date=gets.chomp
-    if !Validation.validate_date(date)
-      return false
-    end
+  # modify event
+  def modify
 
-    p "Enter Event Title:"
-    title=gets.chomp
-    if Validation.check_string(title)
-      return false
-    end
+    user_input = input_for_modify
+    return false unless user_input
+
+    date, title, new_title, new_desc = user_input
+    @total_events.each do |key, value|
+      next unless key == date
+
+      value.each do |event|
+        next unless event.title == title
 
-    total_events.each_with_index do |event,index|
-      if event.has_key?(date) &&  event[date][index].title==title
-      puts "Enter NewTitle:"
-      newtitle=gets.chomp
-      if Validation.check_string(newtitle)
-        return false
+        event.title = new_title
+        event.desc = new_desc
+        p 'Event Modify Successfully'
+        p @total_events
+        return true
       end
-      puts "Enter NewDesc:"
-      newdesc=gets.chomp
-      if Validation.check_string(newdesc)
-        return false
+      p 'No Event match to your date or title'
+    end
+    false
+  end
+  # single event on date
+  def single_event(choice)
+
+    user_input = delete_events(choice)
+    return false unless user_input
+
+    date, title = user_input
+
+    @total_events.each do |key, value|
+      next unless key == date
+
+      value.each do |event|
+        next unless event.title == title
+
+        value.pop
+        p 'Event deleted successfully'
+        p @total_events
+        return true
       end
-      event[date][index].title=newtitle
-      event[date][index].desc=newdesc
+    end
+    p 'No event match to your title or date'
+    false
+  end
+  # all events on date
+  def all_events_on_Date(choice)
+
+    user_input = delete_events(choice)
+    return false unless user_input
+
+    date = user_input
+
+    @total_events.each do |key, _value|
+      next unless key == date
+
+      @total_events.tap { |events| events.delete(key) }
+      p 'Events deleted successfully'
```

Comment:
This method should just delete events and return boolean, printing messages should not be a part of this class.